rust/main/lander/src/adapter/chains/aleo/adapter/core/tests.rs

```diff
@@ -1,16 +1,41 @@
+use std::sync::Arc;
 use std::time::Duration;
 
-use hyperlane_aleo::AleoTxData;
-use hyperlane_core::H256;
+use async_trait::async_trait;
+use uuid::Uuid;
 
-use crate::adapter::chains::AleoAdapter;
+use hyperlane_aleo::{AleoProviderForLander, AleoSigner, AleoTxData, FeeEstimate};
+use hyperlane_core::{ChainResult, H256, H512};
+
+use crate::adapter::chains::{AleoAdapter, AleoTxPrecursor};
 use crate::adapter::AdaptsChain;
 use crate::payload::PayloadDetails;
-use crate::{FullPayload, PayloadStatus, PayloadUuid, TransactionStatus};
-
-fn create_test_adapter() -> AleoAdapter {
-    AleoAdapter {
-        estimated_block_time: Duration::from_secs(10),
+use crate::transaction::{Transaction, VmSpecificTxData};
+use crate::{FullPayload, PayloadStatus, PayloadUuid, TransactionStatus, TransactionUuid};
+
+use super::Precursor;
+
+// Mock provider implementation for testing
+// Note: We use a manual implementation instead of mockall because the trait has
+// generic methods with complex trait bounds that mockall cannot handle well.
+struct MockAleoProvider;
```

Comment:
Are we able to combine this with `MockProviderWithError` and just have a single `MockAleoProvider`?

Reply:
I've combined mock which were close to each other. Others, I would say, can stay.